src/strategies/backscratcher/strategy-frax-dai-univ3.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../strategy-univ3-base.sol";
+import "../../interfaces/backscratcher/IStrategyProxy.sol";
+import "hardhat/console.sol";
+
+contract StrategyFraxDaiUniV3 is StrategyUniV3Base {
+    address public strategyProxy;
+
+    address public frax_dai_pool = 0x97e7d56A0408570bA1a7852De36350f7713906ec;
+    address public frax_dai_gauge = 0xF22471AC2156B489CC4a59092c56713F813ff53e;
+
+    address public constant FXS = 0x3432B6A60D23Ca0dFCa7761B7ab56459D9C964D0;
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant FRAX = 0x853d955aCEf822Db058eb8505911ED77F175b99e;
+
+    address[] public rewardTokens = [FXS, DAI, FRAX];
+
+    constructor(
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    ) public StrategyUniV3Base(frax_dai_pool, -50, 50, _governance, _strategist, _controller, _timelock) {}
+
+    // **** Views ****
+
+    //for test only
+    function setStrategyProxy(address _proxy) external {
+        require(msg.sender == governance, "!governance");
+        strategyProxy = _proxy;
+    }
+
+    function getName() external pure override returns (string memory) {
+        return "StrategyFraxDaiUniV3";
+    }
+
+    function harvest() public override onlyBenevolent {
+        IStrategyProxy(strategyProxy).harvest(frax_dai_gauge, rewardTokens);
+
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            console.log("rewardToken=> ", rewardTokens[i]);
+            console.log("balance=> ", IERC20(rewardTokens[i]).balanceOf(address(this)));
+        }
+
+        uint256 _fxs = IERC20(FXS).balanceOf(address(this));
+
+        IERC20(FXS).safeApprove(univ2Router2, 0);
+        IERC20(FXS).safeApprove(univ2Router2, _fxs);
+
+        _swapUniswap(FXS, FRAX, _fxs);
+        console.log("\nAfter swap");
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            console.log("rewardToken=> ", rewardTokens[i]);
+            console.log("balance=> ", IERC20(rewardTokens[i]).balanceOf(address(this)));
+        }
+
+        uint256 _frax = IERC20(FRAX).balanceOf(address(this));
+        uint256 _dai = IERC20(DAI).balanceOf(address(this));
+        uint256 _ratio = getProportion();
+        console.log("_ratio=> ", _ratio);
+        uint256 _amount1Desired = (_dai.add(_frax)).mul(_ratio).div(_ratio.add(1e18));
```

Comment:
Can you please explain how this calculation works? 
- Doesn't this assume that DAI and FRAX are the same price?
- Why is `1e18` added to the `_ratio` in the divisor?

Reply:
Resolved:
```
let's say the ratio is 1.5
which means we need 10 DAI and 15 FRAX to add the liquidity
otherwise there will be a dust amount
but there is 5 dai and 20 frax in the pool
so we need to swap 5 frax to 5 dai

uint256 _amount1Desired = (_dai.add(_frax)).mul(_ratio).div(_ratio.add(1e18));
so amount1Desired = (5+20) * 1.5 / (1.5 + 1) = 15
```